### PR TITLE
fix(#533) Replace Park and Ride icon in itinerary

### LINKED ISSUE #533: Replace Park&Ride symbol
The current symbol for park and ride suggestions is not very meaningfull, as it only displays a "P" and a car:
grafik

HSL's symbology (showing a bicycle and small transit symbol in the background for bike & ride, would be car & transit for P&R) seems much more intuitive:
grafik

HSL probably already has prepared such symbols, so coordinate with them.

As we ae currently preparing Flex&Transit routing, a resembling symbol, perhaps with a taxi symbol (e.g. a car with a black bar on top) combined with transit instead of a car symbol should be prepared as well.

### PR DESCRIPTION
**Pull request by jpidelatorre**
- Use the same logic for the park and ride composite icon as the one used for bike and vehicle icons.
- Apply an equivalent change to the itinerary icon for park and ride.

Closes #533

### CODE REVIEW

**Comment by jpidelatorre** [comment]
## Result

image

image

**Comment by leonardehrenfried** [comment]
In your example the blue upper icon should be the bus.

**Comment by leonardehrenfried** [comment]
Compare with bike and ride:
image

**Comment by leonardehrenfried** [comment]
In the summary it doesn't need to show a the car twice, just like in the bike view.

Screenshot from 2021-06-07 18-09-29

**Comment by jpidelatorre** [comment]
I was not filtering the steps correctly. It's done now.

image

**Comment by jpidelatorre** [comment]
> In the summary it doesn't need to show a the car twice, just like in the bike view.
> 
> Screenshot from 2021-06-07 18-09-29

Something like this?
image

**Comment by leonardehrenfried** [comment]
what happens when it's just driving? https://dev.stadtnavi.eu/reiseplan/Nufringen%3A%3A48.6224388%2C8.8881021/Herrenberg%3A%3A48.595557%2C8.8674715/parkAndRide?time=1623092264

**Comment by jpidelatorre** [comment]
> what happens when it's just driving? https://dev.stadtnavi.eu/reiseplan/Nufringen%3A%3A48.6224388%2C8.8881021/Herrenberg%3A%3A48.595557%2C8.8674715/parkAndRide?time=1623092264

It looks like this:

image

**Comment by leonardehrenfried** [comment]
And the icon on top?

**Comment by jpidelatorre** [comment]
It looks like this:

image

**Comment by leonardehrenfried** [comment]
Then we have no way to tell the difference between drive and park + walk. What does it look like if you put the walk symbol in the background?

**Comment by leonardehrenfried** [comment]
Can you please put the walking icon behind the car when it's just car + walk?

**Comment by jpidelatorre** [comment]
I could (same with bike), but there's always a walk before a train or bus. That would mean the other icons would never appear.

**Comment by leonardehrenfried** [comment]
Well, can't you modify the logic to check if it's _only_ drive and walk?

**Comment by jpidelatorre** [comment]
All the way up to the destination? Sure.

**Comment by leonardehrenfried** [comment]
Yes, you can get all the modes from the itinerary and it it's just WALK and CAR then display the walking person icon. Maybe it looks ugly though so fake it before writing the logic.

**Comment by jpidelatorre** [comment]
It looks like this:

image

**Comment by leonardehrenfried** [comment]
It would probably look nicer with a different coloured background like the rest but we can ask the designers for that. Let's do it.

**Comment by leonardehrenfried** [comment]
One question: this is only for the Park and Ride button, right?

**Comment by jpidelatorre** [comment]
Yes, only for park and ride.

**Comment by jpidelatorre** [comment]
As per @hbruch request, here's how the new icon looks like:

image

File: app/component/SummaryRow.js [hunk 1]
```diff
@@ -72,6 +72,7 @@ export const RouteLeg = ({
   isTransitLeg,
   fitRouteNumber,
   withBicycle,
+  subIcon,
```
Comment (leonardehrenfried) [inline app/component/SummaryRow.js]: Is the subicon required in this component?

File: app/component/RouteNumber.js [hunk 2]
```diff
@@ -76,6 +79,9 @@ function RouteNumber(props, context) {
             className="itinerary-icon_with-bicycle"
           />
         )}
+        {subIcon && (
```
Comment (leonardehrenfried) [inline app/component/RouteNumber.js]: what does this do?

**Comment by leonardehrenfried** [comment]
The colours of train trips are also messed up:
### Before
Screenshot from 2021-06-09 13-23-02
### After
Screenshot from 2021-06-09 13-23-08

**Comment by leonardehrenfried** [comment]
Screenshot from 2021-06-09 13-25-53

This display is also incorrect. Please calculate the distance for the driving part.

**Comment by jpidelatorre** [comment]
@leonardehrenfried  I pushed some corrections. I think every point is solved with that.

**This is what I see in the icons. Is this correct?**

image

---

**Is this the correct calculation?**

image

**Comment by leonardehrenfried** [comment]
Screenshot from 2021-06-10 11-11-44

Car is missing from P+R.

**Comment by jpidelatorre** [comment]
image

Fixed. I had removed the color for the icon from the markup without adding it back in the stylesheet. Now it is correct.

**Comment by leonardehrenfried** [comment]
Thanks.

**Pull request merged**